engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/BooleanPredicateClausesStep.java

```diff
@@ -16,4 +16,8 @@
 public interface BooleanPredicateClausesStep<S extends BooleanPredicateClausesStep<?>>
 		extends GenericBooleanPredicateClausesStep<S, BooleanPredicateOptionsCollector<?>> {
 
+	/**
+	 * @return {@code true} if boolean predicate has at least one clause added, {@code false} otherwise.
+	 */
+	boolean hasClause();
```

Comment:
I think you can remove this, since `BooleanPredicateClausesStep` extends `BooleanPredicateOptionsCollector`.

```suggestion
```